### ISSUE TITLE
Allow setting BindExecutionOrder from ZenjectBinding component

### ISSUE THREAD

**Issue by mysticfall** [ssannandeji/Zenject-2019#365]
Currently, the execution order of components that are defined with `ZenjectBinding` depends on the order of their parent game objects in the scene hierarchy.

Sometimes, it is desirable to arrange such game objects differently from the order they should be initialized (i.e. in alphabetical order). In that case, allowing to set `BindExecutionOrder` via a property of `ZenjectBinding` could be useful.

**Comment by svermeulen** [comment]
But what if you have multiple instances of the same MonoBehaviour in your scene, with different values for this ExecutionOrder property?  (right now, execution order can only be set by type not instance)

**Comment by mysticfall** [comment]
Sorry, it seems I hadn't thought about the problem thoroghly before raising the issue.

By the way, is there any conceptual limitation as to allow specifying the execution order per instance? If not, maybe we could revisit the issue when/if the situation with the execution order will be changed later?

**Comment by svermeulen** [comment]
No, there isn't any conceptual limitation.  We could add support for a per-instance execution order.  That would make sense as a feature and not be too difficult to support

**Comment by mysticfall** [comment]
That's a good news for me, and thanks for the clarification :smile:
